### PR TITLE
Adds events to internals page

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/10781

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
- enable rewards
- go to internals page
- go to event logs tab
- make sure that you see data in there
- go to rewards page
- disable AC
- go back to internals page
- make sure that you see new records saying that ac is 0

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/state/state.cc [hunk 1]
```diff
@@ -240,16 +272,23 @@ std::vector<uint8_t> State::GetRecoverySeed() {
 }
 
 void State::SetRecoverySeed(const std::vector<uint8_t>& seed) {
+  const std::string seed_string = base::Base64Encode(seed);
+  std::string event_string;
+  if (seed.size() > 1) {
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/state/state.cc:277]: Should this be a `DCHECK` or is a seed size <= 1 also valid?

File: components/brave_rewards/browser/event_log.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_REWARDS_BROWSER_EVENT_LOG_H_
+#define BRAVE_COMPONENTS_BRAVE_REWARDS_BROWSER_EVENT_LOG_H_
+
+#include <string>
+
+namespace brave_rewards {
+
+struct EventLog {
```
Comment (zenparsing) [inline components/brave_rewards/browser/event_log.h:13]: naming: Since this is an entry in an event log, maybe `EventLogEntry`?

File: components/definitions/rewardsInternals.d.ts [hunk 3]
```diff
@@ -68,4 +69,11 @@ declare namespace RewardsInternals {
     address: string
     status: WalletStatus
   }
+
+  export interface EventLog {
```
Comment (zenparsing) [inline components/definitions/rewardsInternals.d.ts:73]: naming: EventLogEntry?

File: vendor/bat-native-ledger/include/bat/ledger/public/interfaces/ledger.mojom [hunk 4]
```diff
@@ -535,3 +535,9 @@ enum InlineTipsPlatforms {
   GITHUB = 3
 };
 
+struct EventLog {
```
Comment (zenparsing) [inline vendor/bat-native-ledger/include/bat/ledger/public/interfaces/ledger.mojom:538]: naming EventLogEntry?

File: vendor/bat-native-ledger/src/bat/ledger/internal/database/database_event_log.cc [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  const std::string query = base::StringPrintf(
+      "INSERT INTO %s (event_log_id, key, value, created_at) "
+      "VALUES (?, ?, ?, ?)",
+      kTableName);
+
+  auto command = ledger::DBCommand::New();
+  command->type = ledger::DBCommand::Type::RUN;
+  command->command = query;
+
+  BindString(command.get(), 0, base::GenerateGUID());
```
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/database/database_event_log.cc:52]: Do we need a GUID field? Will the row ever be correlated with any other data?

File: vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc [hunk 6]
```diff
@@ -769,4 +769,8 @@ void LedgerImpl::Shutdown(ledger::ResultCallback callback) {
   });
 }
 
+void LedgerImpl::GetEventLogs(ledger::GetEventLogsCallback callback) {
+  database()->GetAllEventLogs(callback);
```
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc:773]: Is there any danger of returning too much data here?

File: vendor/bat-native-ledger/src/bat/ledger/internal/state/state.cc [hunk 7]
```diff
@@ -240,16 +272,23 @@ std::vector<uint8_t> State::GetRecoverySeed() {
...
+    event_string = std::to_string(seed[0]+seed[1]);
```
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/state/state.cc:278]: Does `seed.substr(0, 2)` work?

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.cc [hunk 8]
```diff
@@ -209,6 +210,10 @@ void Uphold::DisconnectWallet() {
     return;
   }
 
+  ledger_->database()->SaveEventLog(
+      ledger::log::kWalletDisconnected,
+      static_cast<std::string>(ledger::kWalletUphold) + "/" + wallet->address);
```
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.cc:215]: nit: could probably also just do `std::string(ledger::kWalletUphold)` if preferred.

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_authorization.cc [hunk 9]
```diff
@@ -238,6 +239,10 @@ void UpholdAuthorization::OnCardCreate(
       ledger::kWalletUphold,
       wallet_ptr->Clone());
 
+  ledger_->database()->SaveEventLog(
+      ledger::log::kWalletConnected,
+      static_cast<std::string>(ledger::kWalletUphold) + "/" + address);
```
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_authorization.cc:244]: nit: could probably also just do `std::string(ledger::kWalletUphold)` if preferred.

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc [hunk 10]
```diff
@@ -143,6 +144,10 @@ void Wallet::DisconnectWallet(
     return;
   }
 
+  ledger_->database()->SaveEventLog(
+      ledger::log::kWalletDisconnected,
+      static_cast<std::string>(wallet_type) + "/" + wallet_ptr->address);
```
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc:149]: nit: could probably also just do `std::string(ledger::kWalletUphold)` if preferred.

File: vendor/bat-native-ledger/src/bat/ledger/internal/database/database_activity_info_unittest.cc [hunk 11]
```diff
@@ -177,28 +186,4 @@ TEST_F(DatabaseActivityInfoTest, DeleteRecordEmpty) {
   activity_->DeleteRecord("", [](const ledger::Result){});
 }
 
-TEST_F(DatabaseActivityInfoTest, DeleteRecordOk) {
```
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/database/database_activity_info_unittest.cc:180]: This removal seems unrelated - was it intentional?

File: vendor/bat-native-ledger/src/bat/ledger/internal/state/state.cc (hunk 1 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/state/state.cc:277]: in theory seed can get corrupted, so we need this check

File: vendor/bat-native-ledger/src/bat/ledger/internal/state/state_migration_v4.cc [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+  // Auto contribute
+  auto enabled = ledger_->ledger_client()->GetBooleanState(
+      ledger::kStateAutoContributeEnabled);
+  ledger_->database()->SaveEventLog(
+      ledger::kStateAutoContributeEnabled,
+      std::to_string(enabled));
+
+  // Seed
+  if (seed.size() > 1) {
+    const std::string event_string = std::to_string(seed[0]+seed[1]);
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/state/state_migration_v4.cc:37]: done

File: components/brave_rewards/browser/event_log.h (hunk 2 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/event_log.h:13]: Not sure if `Entry` adds additional value

File: components/definitions/rewardsInternals.d.ts (hunk 3 above)
Comment (NejcZdovc) [reply components/definitions/rewardsInternals.d.ts:73]: same as above

File: vendor/bat-native-ledger/include/bat/ledger/public/interfaces/ledger.mojom (hunk 4 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/include/bat/ledger/public/interfaces/ledger.mojom:538]: same as above

File: vendor/bat-native-ledger/src/bat/ledger/internal/database/database_event_log.cc (hunk 5 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/database/database_event_log.cc:52]: currently no, but could be

File: vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc (hunk 6 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc:773]: limited to 2000 records

File: vendor/bat-native-ledger/src/bat/ledger/internal/state/state.cc (hunk 7 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/state/state.cc:278]: fixed

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.cc (hunk 8 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.cc:215]: `static_cast<std::string>` is approach that is defined in chromium style guides

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_authorization.cc (hunk 9 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_authorization.cc:244]: same as above

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc (hunk 10 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc:149]: same as above

File: vendor/bat-native-ledger/src/bat/ledger/internal/database/database_activity_info_unittest.cc (hunk 11 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/database/database_activity_info_unittest.cc:180]: reverted

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.cc [hunk 13]
```diff
@@ -208,6 +209,13 @@ void Uphold::DisconnectWallet() {
     return;
   }
 
+  if (!wallet->address.empty()) {
+    ledger_->database()->SaveEventLog(
+      ledger::log::kWalletDisconnected,
```
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.cc:214]: Nit: indent spacing

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc [hunk 14]
```diff
@@ -144,6 +145,13 @@ void Wallet::DisconnectWallet(
     return;
   }
 
+  if (!wallet_ptr->address.empty()) {
+    ledger_->database()->SaveEventLog(
+      ledger::log::kWalletDisconnected,
```
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc:150]: Nit: indent spacing

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.cc (hunk 13 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.cc:214]: fixed

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc (hunk 14 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc:150]: fixed

**Comment by NejcZdovc** [comment]
Ci failed on known browser tests
